iris-mpc-gpu/src/server/actor.rs

```diff
@@ -1565,6 +1586,37 @@ impl ServerActor {
 
         Ok((compact_device_queries, compact_device_sums))
     }
+
+    fn populate_or_policy_bitmap(&mut self, or_rule_serial_ids: Vec<Vec<u32>>, batch_size: usize) {
+        for (device_idx, &db_length) in self.current_db_sizes.iter().enumerate() {
+            if db_length == 0 {
+                continue; // Skip empty DBs
+            }
+
+            let row_stride64 = (db_length + 63) / 64;
+            let total_size = row_stride64 * batch_size / ROTATIONS;
+
+            // Create the bitmap on the host
+            let mut bitmap = vec![0u64; total_size];
+
+            for (query_idx, db_indices) in or_rule_serial_ids.iter().enumerate() {
+                for &db_idx in db_indices {
+                    if db_idx < db_length as u32 {
+                        let row_start = query_idx * row_stride64;
+                        let word_idx = row_start + (db_idx as usize / 64);
+                        let bit_offset = db_idx as usize % 64;
+                        bitmap[word_idx] |= 1 << bit_offset;
+                    }
+                }
+            }
+
+            // Copy the host-side bitmap into the pre-allocated GPU memory
+            self.device_manager
+                .device(device_idx)
+                .htod_copy_into(bitmap, &mut self.or_policy_bitmap[device_idx])
+                .expect("Failed to populate OR policy bitmap");
+        }
```

Comment:
I think you need here to htod copy on the same as the join (`streams[0]`). You can use this helper for it: `htod_on_stream_sync`.
Otherwise you might not await it.

Reply:
can `htod_sync_copy_into` instead be used?